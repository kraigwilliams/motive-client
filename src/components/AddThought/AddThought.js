import React, { Component } from 'react';

import Label from '../../components/Utils/Label';
import Dropdown from '../../components/Utils/Dropdown';
import Textarea from '../../components/Utils/Textarea';
import Button from '../../components/Utils/Button';
import Input from '../../components/Utils/Input';


export default class AddThought extends Component {

  render() {
    return (
      <div>
        {/* <Title>Create A Thought</Title> */}
        
        <Label>Title</Label>
        <Input />

        <Label>Type</Label>
        <Dropdown />

        <Label>Topic</Label>
        <Dropdown />

        <Label>Content</Label>
        <Textarea />

        <Button type='submit'>Submit</Button>

      </div>
    );
  }
}

