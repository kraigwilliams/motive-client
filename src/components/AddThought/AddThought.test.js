import React from 'react'
    import ReactDOM from 'react-dom'
    import AddThought from './AddThought'
    import renderer from 'react-test-renderer';
    
    describe(`<AddThought />`, () => {
      //Smoke Testing 
      it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
          <AddThought />,
          div
        )
        ReactDOM.unmountComponentAtNode(div)
      })
      //Snapshot Testing
      it('renders UI as expected', () => {
        const tree = renderer 
          .create(<AddThought />)
          .toJSON()
        expect(tree).toMatchSnapshot()
      })
    })

