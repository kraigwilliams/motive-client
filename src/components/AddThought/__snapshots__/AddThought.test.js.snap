// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddThought /> renders UI as expected 1`] = `
<div
  className="sc-AxjAm jsqeJd"
>
  <form
    className="sc-AxmLO ghXbMg"
    onSubmit={[Function]}
  >
    <button
      className="Button sc-AxiKw eJdQFo"
      margin="0px"
      onClick={[Function]}
      type="reset"
    />
    <h1
      className="sc-Axmtr nbDTi"
    >
      Create a Thought
    </h1>
    <label
      className="Label sc-fzoLsD hJqNJg"
      htmlFor="thought-title"
    >
      Title
    </label>
    <input
      aria-label="Enter title of this thought"
      aria-required="true"
      className="Input sc-fzpans jwpxrB"
      id="thought-title"
      name="title"
      required={true}
      type="text"
    />
    <label
      className="Label sc-fzoLsD hJqNJg"
      htmlFor="thought-topic"
    >
      Topic
    </label>
    <select
      aria-label="You can select a topic to put this thought in"
      className="sc-fznyAO kbbavt"
      id="thought-topic"
      name="topic"
      onChange={[Function]}
      value={0}
    >
      <option
        value={0}
      >
         -- Not in a Topic -- 
      </option>
    </select>
    <label
      className="Label sc-fzoLsD hJqNJg"
      htmlFor="thought-content"
    >
      Content
    </label>
    <textarea
      aria-label="Create the content of this thought"
      aria-required="true"
      className="Textarea sc-fzplWN cJcilO"
      id="thought-content"
      name="content"
      required={true}
    />
    <button
      className="Button sc-AxhUy dpUUoP"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
